providers/implementations/signature/ml_dsa_sig.c

```diff
@@ -281,12 +404,26 @@ static int ml_dsa_set_ctx_params(void *vctx, const OSSL_PARAM params[])
     if (p != NULL && !OSSL_PARAM_get_int(p, &pctx->mu))
         return 0;
 
+    if (pctx->operation == EVP_PKEY_OP_VERIFYMSG) {
+        p = OSSL_PARAM_locate_const(params, OSSL_SIGNATURE_PARAM_SIGNATURE);
+        if (p != NULL) {
+            OPENSSL_free(pctx->sig);
+            pctx->sig = NULL;
+            pctx->siglen = 0;
+            if (!OSSL_PARAM_get_octet_string(p, (void **)&pctx->sig,
+                                             0, &pctx->siglen))
+                return 0;
+        }
+    }
+
     return 1;
 }
 
 static const OSSL_PARAM *ml_dsa_settable_ctx_params(void *vctx,
                                                     ossl_unused void *provctx)
 {
+    PROV_ML_DSA_CTX *pctx = (PROV_ML_DSA_CTX *)vctx;
+
     static const OSSL_PARAM settable_ctx_params[] = {
         OSSL_PARAM_octet_string(OSSL_SIGNATURE_PARAM_CONTEXT_STRING, NULL, 0),
```

Comment:
Maybe macro the shared ones as a #define and include them in both..

Reply:
ok (boy do I hate all these macros, makes reading the code sooo hard)